Azure-Samples/aks-store-demo
# refactor: re-write virtual-worker using rust

## Purpose

* Re-write virtual-worker using rust and call makeline-service instead of reading from rabbitmq and writing to mongodb directly
* Updates to aks-store-all-in-one.yaml to remove redis and add virtual-customer

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code then run the following

```
docker compose up
```

* Test the code

Run the following command to pull orders from the RabbitMQ, dump into Redis, then process into MongoDB

```bash
curl http://localhost:3001/fetch
```

Run the following command to get an order for processing

```bash
curl http://localhost:3001/order/:id # where :id is the order id
```

Run the following command to complete the processing of an order

```bash
curl -X PUT -H "Content-Type: application/json" -d '{"orderId":"<YOUR_ORDER_ID>","customerId":"1022466235","items":[{"productId":0,"quantity":2,"price":22.428429},{"productId":0,"quantity":1,"price":99.63117},{"productId":0,"quantity":2,"price":93.84932}],"status":1}' http://localhost:3001/order
```

> NOTE: Make sure to update `<YOUR_ORDER_ID>` with an existing order ID in the JSON payload

## What to Check
Verify that the following are valid
* Now you can run the following command to get the order from MongoDB

```bash
# connect to mongodb
mongosh

# show databases and confirm orderdb exists
show dbs

# use orderdb
use orderdb

# show collections and confirm orders exists
show collections

# get the order and confirm status
db.orders.findOne({orderid: '<YOUR_ORDER_ID>'})
```

> NOTE: Make sure to update `<YOUR_ORDER_ID>` with an existing order ID in the JSON payload
## Other Information
